allennlp/training/metrics/mention_recall.py

```diff
@@ -0,0 +1,33 @@
+from overrides import overrides
+
+from allennlp.training.metrics.metric import Metric
+
+
+@Metric.register("mention_recall")
+class MentionRecall(Metric):
+    def __init__(self) -> None:
+        self._num_gold_mentions = 0
+        self._num_recalled_mentions = 0
+
+    @overrides
+    def __call__(self, batched_top_spans, batched_metadata):
+        for top_spans, metadata in zip(batched_top_spans.data.tolist(), batched_metadata):
+            gold_mentions = set(m for c in metadata["clusters"] for m in c)
+            top_spans = set(tuple(s) for s in top_spans)
+            self._num_gold_mentions += len(gold_mentions)
+            self._num_recalled_mentions += len(gold_mentions & top_spans)
```

Comment:
It's not clear to me what exactly is getting compared in this `&` call.  Maybe types on the inputs would help?  Or just some docstrings?

Reply:
It's comparing pairs of indices in the form of tuples. I've added some types.